### PR TITLE
DefaultPushServerConfigurationResponse Tests

### PR DESCRIPTION
**Pull request by JKereliuk**
Tests for DefaultPushServerConfigurationResponse

### CODE REVIEW

**Comment by JKereliuk** [comment]
I shouldn't have branch from where I did, please ignore all commits except the last 3

File: components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerConfigurationResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        // populate a JSONObject
+        for(int i = 0; i < 10; i++) {
+            serverResponse.accumulate("accepted_fields", i);
+        }
+        Set<String> expected = new TreeSet<String>();
+        for(int i = 0; i < 10; i++) {
+            expected.add(String.valueOf(i));
+        }
+        Assert.assertEquals(expected, defaultPushServerConfigurationResponse.getRemoteAcceptedPatientFields());
```
Comment (sdumitriu) [inline components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerConfigurationResponseTest.java]: Shouldn't this call `getPushableFields`?

**Comment by marta-** [comment]
@allasm @veronikaslc can you continue the work on these Push tests as part of the Push improvements?

File: components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerConfigurationResponseTest.java (hunk 1 above)
Comment (veronikaslc) [inline components/patient-data-sharing/push-api/src/test/java/org/phenotips/data/push/internal/DefaultPushServerConfigurationResponseTest.java]: this test is an exact copy of `getRemoteAcceptedPatientFieldsTest`, I guess the intention was to implement alike

**Comment by veronikaslc** [comment]
Closed, the work moved and continued in https://github.com/phenotips/phenotips/tree/PT-2044

**Pull request closed without merging**